lib/App/MintTag.pm

```diff
@@ -493,6 +495,42 @@ sub _maybe_delete_source_branches ($self, $step) {
   }
 }
 
+sub _maybe_cleanup_tags ($self, $step) {
+  return unless ($step->push_tag_to && $step->cleanup_tag_days);
+
+  my $remote = $step->push_tag_to;
+  my $prefix = $step->tag_prefix;
+
+  try {
+    my $tag_spec = "refs/tags/${prefix}*";
+    my $tag_list = run_git('for-each-ref', "--format=%(refname:short)\t%(creatordate:unix)", $tag_spec);
+
+    my @tags = map {; my ($tag, $ts) = split /\t/ ; { ts => $ts, 'tag' => $tag}  }
+    split /\r?\n/, $tag_list;
+
+    my $time = time;
+
+    my $max_ts = $time - ($step->cleanup_tag_days * 86400);
+
+    for my $chunk (@tags) {
+      if ($chunk->{ts} < $max_ts) {
+        $Logger->log(["Deleting old tag %s", $chunk->{tag}]);
+        run_git('push', $remote->name, ":refs/tags/$chunk->{tag}"  );
+      } else {
+        $Logger->log(["Keeping old tag %s", $chunk->{tag}]);
+      }
+    }
+  } catch {
+    my $err = $_;
+
+    $Logger->log_fatal([
+        "could not delete remote tag on remote %s: %s",
+        $remote->name,
+        $err,
+      ])
+  };
```

Comment:
This is a funny case to catch.  You're saying "could not delete remote tag" but if you had 10 to delete, you may have deleted 5.  It's not necessarily better or worse in the all-at-once approach.  Maybe there's not much to do here other than say "error encountered while deleting tags from remote".  More accurate, but it'll still be a bummer…